<launch>
<node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 base_link olelidar 10" />
<node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0 0 0.0 odom base_link 10"/>
  <arg name="pkg" value="olelidar"/>

  <!-- driver -->
  <arg name="driver" default="true"/>
  <arg name="device_ip" default="192.168.1.100"/>
  <arg name="device_port" default="2368"/>
  <!-- decoder -->
  <arg name="r_max" default="30000"/>
  <arg name="ang_start" default="-180"/>
  <arg name="ang_end" default="180"/>
  <arg name="decoder" default="true"/>
  <arg name="route" default="1600"/>
  <arg name="step" default="0.225"/> 
  <arg name="frame_id" default="olelidar"/>
  <arg name="debug" default="false"/>
  <env if="$(arg debug)" name="ROSCONSOLE_CONFIG_FILE" value="$(find olelidar)/launch/debug.conf"/>

  <group ns="$(arg frame_id)">
    <include file="$(find olelidar)/launch/driver.launch" if="$(arg driver)">
      <arg name="device_ip" value="$(arg device_ip)"/>
      <arg name="device_port" value="$(arg device_port)"/>
    </include>

    <include file="$(find olelidar)/launch/decoder.launch" if="$(arg decoder)">
      <arg name="frame_id" value="$(arg frame_id)"/>
      <arg name="r_max" value="$(arg r_max)"/>
      <arg name="ang_start" value="$(arg ang_start)"/>
      <arg name="ang_end" value="$(arg ang_end)"/>
    </include>
  </group>

</launch>
